webview-ui/src/App.tsx

```diff
@@ -11,42 +11,54 @@ type LoadDataCommand = {
   lines: string;
 };
 
-type AppState = {
-  state: "loading"
-} | {
-  state: "ready";
-  report: Report;
-  property: string | null;
+type AppState =
+  | {
+      state: "loading";
+    }
+  | {
+      state: "ready";
+      report: Report;
+      property: string | null;
+    };
+
+type AppProps = {
+  dataSourceURL?: string;
 };
 
-const App = () => {
+const App = (props: AppProps) => {
   const [state, setStateRaw] = useState<AppState>({ state: "loading" });
   const [shouldShowExplainer, setShouldShowExplainer] = useState<boolean>(true);
 
-  const setState = (newState: AppState) => {
```

Comment:
Had to wrap this and `loadData` in `useCallback` to avoid constantly triggering the effect.

Reply:
Though mostly it's to appease the linter. In practice, the effect will only run a few times until AppState stabilizes and stops changing for the entire lifetime of the app.